src/pages/Animal/MyPets/index.tsx

```diff
@@ -1,42 +1,112 @@
-import { useNavigation } from '@react-navigation/native';
+import React, { useLayoutEffect } from 'react';
 import { setStatusBarBackgroundColor } from 'expo-status-bar';
-import React, { useEffect, useLayoutEffect, useState } from 'react';
-import { Text } from 'react-native';
+
 import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
+
+import { useNavigation } from '@react-navigation/native';
+
 import { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
-import { v4 as uuidv4 } from 'uuid';
-import { AnimalCard } from '../../../components/AnimalCard';
-import { Theme } from '../../../constants';
+
 import HeaderLayout from '../../../layouts/HeaderLayout';
-import { Container } from './styles';
+
+import AnimalCard from '../../../components/AnimalCard';
+import InfiniteScroll from '../../../components/InfiniteScroll';
+
+import { Theme } from '../../../constants';
+
+import { Animal } from '../../../types/animal';
+
+import {
+  CardText, CardTextContainer, CardTextRow, Container,
+} from './styles';
 
 // Service imports.
 import animalAPI from '../../../services/animal/api';
 import userAPI from '../../../services/user/api';
 
 const MyPets = (): JSX.Element => {
+  const user = userAPI.currentUserDocument();
   const navigation = useNavigation();
 
-  const [fetchedPets, setFetchedPets] = useState<FirebaseFirestoreTypes.DocumentData[]>([]);
-
   useLayoutEffect(() => {
     setStatusBarBackgroundColor(Theme.elements.statusBarPrimaryDark, false);
   }, [navigation]);
 
-  const fetchPets = (): void => {
-    const user = userAPI.currentUserDocument();
+  const animalKey = (animalItem: Animal): string => animalItem.id;
+
+  const formatAnimal = (pet: Animal): JSX.Element => (
+    <AnimalCard
+      key={pet?.id}
+      imageUrl={`${pet?.pictures.length > 0 ? `${pet?.pictures[0]}` : null}`}
+      body={(
+        <CardTextContainer>
+          <CardTextRow>
+            <CardText>0 novos interessados</CardText>
+          </CardTextRow>
+        </CardTextContainer>
+      )}
+      title={pet?.name}
+      headerOptions={(
+        <MaterialCommunityIcons
```

Comment:
This is not correct. The right icon should be an "error" icon (exclamation inside circle) if and only if there are new people interested in the pet. That is why the interests API is kinda of needed to complete this card.